docs/developer-docs/smart-contracts/advanced-features/http-request.mdx

```diff
@@ -0,0 +1,201 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Serving an HTTP request
+
+## Overview 
+
+Canisters can serve or handle an incoming HTTP request using the [HTTP Gateway Protocol](https://internetcomputer.org/docs/current/references/http-gateway-protocol-spec). 
+
+This allows developers to host decentralized APIs from a canister.
+
+## How it works
+
+An HTTP request from a client gets intercepted by the HTTP Gateway Protocol, which identifies the target canister and encodes the request in Candid. This encoded request is then sent to the canister for processing. Once processed, the canister replies with an HTTP response. Finally, the HTTP Gateway Protocol decodes the response using Candid and sends it back to the client, completing the communication loop.
+
+For detailed information on how it works, please refer to the [HTTP Gateway Protocol specification](https://internetcomputer.org/docs/current/references/http-gateway-protocol-spec). 
+
+## How to make a request
+
+The following example returns 'Hello, World!' in the body at the `/hello` endpoint. 
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko">
+
+```motoko
+import Buffer "mo:base/Buffer";
+
+type HeaderField = (Text, Text);
+
+type HttpResponse = {
+    status_code: Nat16;
+    headers: [HeaderField];
+    body: Blob;
+};
+
+type HttpRequest = {
+    method: Text;
+    url: Text;
+    headers: [HeaderField];
+    body: Blob;
+  };
+
+public query func http_request(req: HttpRequest): async (HttpResponse) {
+    let path = removeQuery(req.url);
+    if(path == "/hello") {
+        return {
+            status_code = 200,
+            headers = [],
+            body = Buffer.from('Hello, World!'),
+        };
+    };
+        return {
+            body = Text.encodeUtf8("404 Not found :");
+            headers = [];
+            status_code = 404;
+    };
+};
+```
+
+</TabItem>
+
+<TabItem value="rust" label="Rust">
+
+```rust
```

Comment:
Seems like we're missing the definition of `HeaderField` for the rust example. I can see it in both Motoko and Typescript.